pints/_nested/__init__.py

```diff
@@ -50,12 +230,424 @@ def __init__(self, log_likelihood, log_prior):
+                ok = False
+            if not ok:
+                raise ValueError(
+                    'Given method must extend pints.NestedSampler.'
+                )
+
+        self._sampler = method(log_prior=self._log_prior)
+        # Check if sensitivities are required
+        self._needs_sensitivities = self._sampler.needs_sensitivities()
+
+    def set_parallel(self, parallel=False):
+        """
+        Enables/disables parallel evaluation.
+
+        If ``parallel=True``, the method will run using a number of worker
+        processes equal to the detected cpu core count. The number of workers
+        can be set explicitly by setting ``parallel`` to an integer greater
+        than 0.
+        Parallelisation can be disabled by setting ``parallel`` to ``0`` or
+        ``False``.
+        """
+        if parallel is True:
+            self._parallel = True
+            self._n_workers = pints.ParallelEvaluator.cpu_count()
+        elif parallel >= 1:
+            self._parallel = True
+            self._n_workers = int(parallel)
+        else:
+            self._parallel = False
+            self._n_workers = 1
+
+    def parallel(self):
+        """
+        Returns the number of parallel worker processes this routine will be
+        run on, or ``False`` if parallelisation is disabled.
+        """
+        return self._n_workers if self._parallel else False
+
+    def set_marginal_log_likelihood_threshold(self, threshold):
+        """
+        Sets threshold for determining convergence in estimate of marginal
+        log likelihood which leads to early termination of the algorithm.
+        """
+        if threshold <= 0:
+            raise ValueError('Convergence threshold must be positive.')
+        self._marginal_log_likelihood_threshold = threshold
+
+    def marginal_log_likelihood_threshold(self):
+        """
+        Returns threshold for determining convergence in estimate of marginal
+        log likelihood which leads to early termination of the algorithm.
+        """
+        return self._marginal_log_likelihood_threshold
+
+    def iterations(self):
+        """
+        Returns the total number of iterations that will be performed in the
+        next run.
+        """
+        return self._iterations
+
+    def n_posterior_samples(self):
+        """
+        Returns the number of posterior samples that will be returned (see
+        :meth:`set_n_posterior_samples()`).
+        """
+        return self._posterior_samples
+
+    def _initialise_parallel(self, f):
+        """
+        Initialises parallel runners.
+        """
+        # Create evaluator object
+        if self._parallel:
+            # Use at most n_workers workers
+            n_workers = self._n_workers
+            evaluator = pints.ParallelEvaluator(
+                f, n_workers=n_workers)
+        else:
+            evaluator = pints.SequentialEvaluator(f)
+        return evaluator
+
+    def _initialise_sensitivities(self):
```

Comment:
Could this have a name like initialise function, or initialise callable, or something like that? In most cases it won't do sensitivities, I think

Reply:
Done!